app/core/builder/builder.py

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+# For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+from typing import Iterable
+
+from astropy.time import Time
+from lucupy.minimodel import Program
+
+from .blueprint import Blueprint
+from app.core.components.collector import Collector
+from app.core.components.selector import Selector
+from app.core.components.optimizer import Optimizer
+from app.core.scheduler.modes import dispatch_with
+from app.config import config
+
+@dispatch_with(config.mode)
+class SchedulerBuilder:
```

Comment:
Builder is a way to create single components in a single point of reference, this also allows us to use a class decorator (`dispatch_with`) to modify 1 or more than component.